climakitae/tmy.py

```diff
@@ -1,194 +1,346 @@
+
     warming_year_average_range = {
         1.5 : (2034,2063),
         2 : (2047,2076),
         3 : (2061,2090),
-        4 : (2076, 2100)
     }
 
-    def remove_repeats(xr_data):
+    selections.append_historical = False
+    selections.area_average = True
+    selections.resolution = "45 km"
+    selections.scenario = ["SSP 3-7.0 -- Business as Usual"]
+    selections.time_slice = warming_year_average_range[warmlevel]
+    selections.timescale = "hourly"
+    future_da = _read_from_catalog(
+        selections=selections,
+        location=location,
+        cat=catalog
+    )
+
+    # Compute mean of all simulations
+    # Compute zarr data
+    future_da_mean = future_da.mean(dim="simulation").isel(scenario=0).compute()
+    return future_da_mean
+
+
+class AverageMeteorologicalYear(param.Parameterized):
+    """
+    An object that holds the "Data Options" paramters for the
+    explore.tmy panel.
+    """
+
+    units2 = param.ObjectSelector(objects=dict())
+    variable2 = param.ObjectSelector(default="Air Temperature at 2m", objects=dict())
+    cached_area2 = param.ObjectSelector(default="CA", objects=dict())
+    area_subset2 = param.ObjectSelector(default="states", objects=["CA counties", "states"])
+
+    def __init__(self, *args, **params):
+        super().__init__(*args, **params)
+
+        # Selectors defaults
+        self.selections.append_historical = False
+        self.selections.area_average = True
+        self.selections.resolution = "45 km"
+        self.selections.scenario = ["Historical"]  # setting for historical
+        self.selections.time_slice = (1981,2010)
+        self.selections.timescale = "hourly"
+        self.selections.variable = "Air Temperature at 2m"
+
+        self.units2 = self.selections.descrip_dict[self.selections.variable]["native_unit"]
+
+        # Location defaults
+        self.location.area_subset = 'states'
+        self.location.cached_area = 'CA'
+
+        self.param["variable2"].objects = self.selections.param.variable.objects
+        self.param["cached_area2"].objects = self.location.param.cached_area.objects
+
+        # Postage data and anomalies defaults
+        self.historical_tmy_data = _get_historical_tmy_data(
+            selections=self.selections,
+            location=self.location,
+            catalog=self.catalog
+        )
+        self.future_tmy_data = _get_future_heatmap_data(
+            selections=self.selections,
+            location=self.location,
+            catalog=self.catalog,
+            warmlevel=1.5
+        )
+
+    # TMY options to display
+    tmy_options = param.ObjectSelector(default='Absolute',
+        objects=['Absolute', 'Difference'])
+
+    abs_tmy_options = param.ObjectSelector(default='Historical',
+        objects=['Historical', 'Warming Level Future'])
+
+    diff_tmy_options = param.ObjectSelector(default='Warming Level Future',
+        objects=['Warming Level Future', 'Severe AMY'])
```

Comment:
I suggest you remove "Severe AMY" as an option until it is implemented in the code.